File: do-not-share/jads-android/app/src/main/kotlin/com/jads/telemetry/TelemetryPointDto.kt
// T04 — Telemetry point data class for live streaming
package com.jads.telemetry

import kotlinx.serialization.Serializable

@Serializable
data class TelemetryPointDto(
    val missionId: String,
    val uin: String,
    val lat: Double,
    val lon: Double,
    val altAGL: Double,
    val altMSL: Double,
    val speedKmh: Double,
    val headingDeg: Double,
    val batteryPct: Double,
    val satelliteCount: Int,
    val source: String,
    val ts: Long
)

File: jads-ios/Sources/JADS/Telemetry/TelemetryPointDto.swift
// T07 — Telemetry point model for live streaming
import Foundation

struct TelemetryPointDto: Codable {
    let missionId: String
    let uin: String
    let lat: Double
    let lon: Double
    let altAGL: Double
    let altMSL: Double
    let speedKmh: Double
    let headingDeg: Double
    let batteryPct: Double
    let satelliteCount: Int
    let source: String
    let ts: Int64
}

struct ViolationDto: Codable {
    let point: TelemetryPointDto
    let violationType: String
    let distanceToEdge: Double
    let ts: Int64
}
